src/librustdoc/html/render.rs

```diff
@@ -1945,18 +1888,10 @@ fn item_trait(w: &mut fmt::Formatter, cx: &Context, it: &clean::Item,
                   bounds,
                   WhereClause(&t.generics)));
 
-    let types = t.items.iter().filter(|m| {
-        match m.inner { clean::AssociatedTypeItem(..) => true, _ => false }
-    }).collect::<Vec<_>>();
-    let consts = t.items.iter().filter(|m| {
-        match m.inner { clean::AssociatedConstItem(..) => true, _ => false }
-    }).collect::<Vec<_>>();
-    let required = t.items.iter().filter(|m| {
-        match m.inner { clean::TyMethodItem(_) => true, _ => false }
-    }).collect::<Vec<_>>();
-    let provided = t.items.iter().filter(|m| {
-        match m.inner { clean::MethodItem(_) => true, _ => false }
-    }).collect::<Vec<_>>();
+    let types = t.items.iter().filter(|m| m.is_associated_type()).collect(): Vec<_>;
+    let consts = t.items.iter().filter(|m| m.is_associated_const()).collect(): Vec<_>;
+    let required = t.items.iter().filter(|m| m.is_ty_method()).collect(): Vec<_>;
+    let provided = t.items.iter().filter(|m| m.is_method()).collect(): Vec<_>;
```

Comment:
Instead of activating a new unstable feature, could this continue to use `collect::<Vec<_>>()`